TypeSystem/Serialization/test.cc

```diff
@@ -144,6 +202,52 @@ TEST(Serialization, JSON) {
       "{\"j\":43,\"q\":\"bar\",\"v\":[\"one\",\"two\"],\"z\":{\"i\":42,\"s\":\"foo\",\"b\":true,\"e\":100}}",
       JSON(complex_object));
 
+  // Serialization of floating point types.
+  // TODO: Think of dealing with 'NaN' and `inf` - JSON standard does not allow them.
+  //       Their string representation in C++ is platform-specific.
+  //       Thus, this test now is GCC/clang-specific.
+  {
+    static_assert(std::numeric_limits<float>::is_iec559, "`float` type is not IEC-559 compliant.");
+    static_assert(std::numeric_limits<double>::is_iec559, "`double` type is not IEC-559 compliant.");
+
+    WithFloatingPoint floating_min;
+    floating_min.f = std::numeric_limits<float>::min();
+    floating_min.d = std::numeric_limits<double>::min();
+    const std::string serialized_fmin = JSON(floating_min);
+    EXPECT_EQ("{\"f\":1.175494350822287508e-38,\"d\":2.2250738585072013831e-308}", serialized_fmin);
+    WithFloatingPoint fmin_from_json = ParseJSON<WithFloatingPoint>(serialized_fmin);
+    EXPECT_FLOAT_EQ(std::numeric_limits<float>::min(), fmin_from_json.f);
+    // Due to precision issues, minimum double value becomes 0.
```

Comment:
This is the reason for the next line to be `EXPECT_NEAR` instead of `EXPECT_DOUBLE_EQ`.